source/extensions/filters/http/cache/cache_policy.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include "envoy/http/header_map.h"
+#include "envoy/stream_info/filter_state.h"
+
+#include "source/extensions/filters/http/cache/cache_headers_utils.h"
+#include "source/extensions/filters/http/cache/http_cache.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+struct CacheEntryUsability {
+  CacheEntryStatus status = CacheEntryStatus::Unusable;
+  Seconds age = Seconds::max();
+};
+
+class CachePolicyCallbacks {
+public:
+  virtual ~CachePolicyCallbacks() = default;
+
+  virtual const StreamInfo::FilterStateSharedPtr& filterState() PURE;
+};
+
+/**
+ * An extension point for deployment specific caching behavior.
+ */
+class CachePolicy {
+public:
+  virtual ~CachePolicy() = default;
+
+  /**
+   * Calculates the lookup key for storing the entry in the cache.
+   * @param request_headers - headers from the request the CacheFilter is currently processing.
+   */
+  virtual Key createCacheKey(const Http::RequestHeaderMap& request_headers) PURE;
+
+  /**
+   * Modifies the cacheability of the response during decoding.
```

Comment:
"Modifies" is probably a confusing word here IMO. I think we want something along the lines of "Determines the cacheability of response associated with the given request_headers".